sympy/matrices/random.py

```diff
@@ -0,0 +1,2248 @@
+import random
+
+from ..core import I as _i
+from ..core.mul import Mul as _multiply
+from ..functions import sqrt as _sqrt, re as _re, im as _im, \
+    transpose as _t, adjoint as _c
+from ..tensor import shape as _shape
+from .dense import eye as _eye
+
+__all__ = 'identity_matrix', 'projection', 'jordan', 'transposition', \
+          'permutation', 'elementary', 'rotation', 'reflection', \
+          'diagonal_normal', 'jordan_normal', 'isometry_normal', \
+          'triangular', 'invertible', 'singular', \
+          'idempotent', 'nilpotent', 'diagonalizable', 'trigonalizable', \
+          'orthogonal', 'unitary', 'normal', \
+          'symmetric', 'hermite', \
+          'regular_to_singular', 'complex_to_real'
+
+_TEST = False
+
+_eps = 1e-13
+
+
+def _get_value(o, i, j):
+    return o[i, j]
+
+
+def _set_value(o, i, j, v):
+    o[i, j] = v
```

Comment:
What is the purpose of this function? I find the array subscript easier to understand than the function call.

Reply:
Used some of these wrappers to wrap the matrix and complex numbers framework (e.g. like `_inv` or `_re`). This keeps the code agnostic so it can be used for other frameworks (like `numpy`, `mpmath` or other.). 
I would appreciate it if we keep it that way.

Reply:
Can you name a framework that doesn't use `o[i, j] =v` for setting a matrix element?

Reply:
I think `tensorflow` does not support such value assignment. Well yes, `tf` doesn't support any assignment at all. 

But some naive frameworks which simply use lists of lists `[[a,  b], [c, d]]` to represent matrices would take `o[i][j]=v`. Maybe we can use `o[i, j] = v` as you prefer.

Reply:
It's possible to just convert between different matrix formats so it's not clear to me what the value is in needing to support so many rather than just saying that the matrix can be constructed in one format and then converted to another.

Reply:
Since `sympy` is quite slow, creating in `sympy ` and converting it would not be efficient. Nevertheless, I already removed the `_set_value` ;)